std/container/rbtree.d

```diff
@@ -1711,13 +1711,18 @@ assert(equal(rbt[], [5]));
     }
 
     /**
-    Formats the RedBlackTree into a sink function. For more info see
-    $(D std.format.formatValue)
-    */
-    void toString(scope void delegate(const(char)[]) sink, FormatSpec!char fmt) const {
-        sink("RedBlackTree(");
-        sink.formatValue(this[], fmt);
-        sink(")");
+      Formats the RedBlackTree into a sink function. For more info see $(D
+      std.format.formatValue). Note that this only is available when the
+      element type can be formatted. Otherwise, the default toString from
+      Object is used.
+     */
+    static if(is(typeof((){FormatSpec!(char) fmt; formatValue((const(char)[]) {}, ConstRange.init, fmt);})))
```

Comment:
This was a PITA to do. At first, I did `if(is(typeof(formatValue((const(char)[]){}, ConstRange.init, FormatSpec!(char).init))`, but formatValue takes the format spec by ref.

Perhaps there is a nicer trait. I'm not sure.